Fix Depricated Search Endpoint

GamerClassN7 opened the pull request:
<!-- markdownlint-disable MD002 -->
<!-- markdownlint-disable MD041 -->
<!-- Provide a general summary of your changes in the Title above -->

### Description
Fix almost unusable function Get-JiraIssue After last jira api updtate
Issue: https://github.com/AtlassianPS/JiraPS/issues/535

<!-- Describe your changes in detail -->

### Motivation and Context

<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here as follows: -->
<!-- closes #1, closes #2, ... -->

### Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [X] My code follows the code style of this project.
- [X] I have added Pester Tests that describe what my changes should do. **(Not necessary)**
- [X] I have updated the documentation accordingly. **(Not necessary)**

lipkau committed:
setup for alpha release

robertmbaker committed:
Createmeta update (#488)

* Update Get-JiraIssueCreateMetadata.ps1

https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

* Update ConvertTo-JiraCreateMetaField.ps1

https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

* Update Get-JiraIssueCreateMetadata.ps1

https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

* Update ConvertTo-JiraCreateMetaField.Unit.Tests.ps1

Altered test data to match what is returned by new createmeta call.

micheleliberman committed:
BugFix issue #482 allowing Component ID in New-Issue (#483)

Co-authored-by: Michele Liberman <mliberman@smartcommunications.com>

lipkau committed:
Added release notes for 2.15-alpha1

hmmwhatsthisdo committed:
Enable creation of ApplicationData folder if it doesn't exist yet (#513)

hmmwhatsthisdo committed:
Improve performance of `New-`/`Set-JiraIssue` on instances with large quantities of fields (#516)

* Refactor Set-JiraIssue to only call Get-JiraField once

* Reduce log-spam and add explicit debug message before calling Get-JiraField

* Refactor New-JiraIssue to only call Get-JiraField once

* Re-add missing variable declarations

* Reunify parameter formatting in logs

* Fix New-JiraIssue unit tests

* Add mock for Get-JiraField

* Remove unnecessary mock

---------

Co-authored-by: hmmwhatsthisdo <2093321+hmmwhatsthisdo@users.noreply.github.com>

jkc-sw committed:
Add tests for conflicts in New-JiraIssues when fields have same name (#520)

Co-authored-by: Kuanju Chen <kuanju0919@yahoo.com>

hmmwhatsthisdo committed:
Unify Label parameter names to singular form in `New-`/`Set-JiraIssue` (#522)

* Add `-Labels` --> `-Label` alias to `Set-JiraIssue`

* Rename+alias `-Labels` to `-Label` in `New-JiraIssue`

* Update docs

* Update unit tests

---------

Co-authored-by: hmmwhatsthisdo <2093321+hmmwhatsthisdo@users.noreply.github.com>

GamerClassN7 committed:
Update Get-JiraIssue.ps1

harryjohn-dev commented:
@GamerClassN7 This definitely needs fixing but is this the right fix? 

The error in https://github.com/AtlassianPS/JiraPS/issues/535 talks about migrating to the v3 endpoint, so I'd be expecting to see this change frrom v2 to v3:

```powershell
        $searchURi = "$server/rest/api/2/search"
        $resourceURi = "$server/rest/api/2/issue/{0}"
```

Here:

https://github.com/AtlassianPS/JiraPS/blob/339b93e8239c51780bf9ad864cbdf885b67f701c/JiraPS/Public/Get-JiraIssue.ps1#L99

GamerClassN7 commented:
> @GamerClassN7 This definitely needs fixing but is this the right fix?
> 
> The error in #535 talks about migrating to the v3 endpoint, so I'd be expecting to see this change from v2 to v3
> 
> https://github.com/AtlassianPS/JiraPS/blob/339b93e8239c51780bf9ad864cbdf885b67f701c/JiraPS/Public/Get-JiraIssue.ps1#L99

Yes it is i was not able to make it work under api v3
with my fix it seems to work  correctly even pagination etc.

GamerClassN7 committed:
Propper Fix to v3

GamerClassN7 committed:
FIx for last page of iteration

GamerClassN7 commented:
Problem is that this function affect many companies worl wide so maybe as common groud would be possible to make some betta release and get them working and then do the refactoring as you sugested ? :)

GamerClassN7 committed:
Remove incorrect code

lipkau committed:
WIP

GamerClassN7 committed:
Fix Some Errors

GamerClassN7 committed:
Fixes

GamerClassN7 committed:
Remove Write-Host

GamerClassN7 commented:
Probably related Add-JiraIssueWorklog no loger is able to inser worklog item

GamerClassN7 commented:
Also there is soem changes to comment system  in general comment in worklogs now return @{type=doc; version=1; content=System.Object[]} instead of comments.

SrBlackVoid committed:
Updated assertion syntax

SrBlackVoid committed:
Update required Pester module to 5.7.1

SrBlackVoid committed:
Updated PSScriptAnalyzer tests to v5

Also serves as initial basis for how test files may need to be
reformatted.

SrBlackVoid committed:
Updated project-wide test files (minus Help)

SrBlackVoid committed:
Updated Help tests

Tried to preserve the original structure of the tests as much as
possible while converting it to v5 compatibility.

SrBlackVoid committed:
Updated Add-JiraIssueLink Tests (1st function file)

Established overall strategy for updating to Pester v5. Key elements
that may show in subsequent function test updates:
- Removing It blocks from shared functions (causes conflicts with
discovery phase) and adjusting how these functions are called
- Removing InModuleScope usage (not recommended practice)
- Adjusting mock setups & replacing Assert-MockCalled with Should
-Invoke
- Some cases of re-organizing BeforeAll/Each and Context blocks

SrBlackVoid committed:
Updated all Add-* function tests

SrBlackVoid committed:
Updated all Get-* function tests

Leaving out Get-JiraIssueLinkType as a special exception.

SrBlackVoid committed:
Updated Get-JiraIssueLinkType function and tests

Get-JiraIssueLinkType was the only function (so far) that needed some
rework to remove the recursive call, which was making testing extremely
challenging.

SrBlackVoid committed:
Updated all New-* function tests

SrBlackVoid committed:
Updated all Remove-* function tests

SrBlackVoid committed:
Updated all Set-* function tests

SrBlackVoid committed:
Updated all Convert* function tests

Because these are all private functions, took a different route with
InModuleScope usage (still follows recommended guidelines).

SrBlackVoid committed:
Updated misc. function tests

Leaving out Invoke-JiraMethod as exception (special case)

SrBlackVoid committed:
Updated Invoke-JiraMethod tests

This one put up a fight. There should be a way to not require completely
surrounding in InModuleScope, but could not get current test structure
to work without it.

SrBlackVoid committed:
Changes needed to pass project-wide tests

Couple of ScriptAnalyzer tests and a missing mandatory parameter flag.

lipkau committed:
Merge remote-tracking branch 'origin/heads/develop'

lipkau committed:
Merge branch 'release/2.15-alpha'

lipkau committed:
Refactor: Transition to single-branch (master) tag-based release strategy

- Fix release workflow: correct asset name (PSIni.zip → JiraPS.zip)
- Replace deprecated GitHub Actions with softprops/action-gh-release@v2
- Add automatic prerelease detection for alpha/beta/rc tags
- Update documentation to reflect master-only branch strategy
- Remove all references to develop branch workflow

lipkau committed:
Refactor: Update line endings and clean up configuration files

dependabot[bot] committed:
:arrow_up: Bump dawidd6/action-download-artifact from 9 to 11

Bumps [dawidd6/action-download-artifact](https://github.com/dawidd6/action-download-artifact) from 9 to 11.
- [Release notes](https://github.com/dawidd6/action-download-artifact/releases)
- [Commits](https://github.com/dawidd6/action-download-artifact/compare/v9...v11)

---
updated-dependencies:
- dependency-name: dawidd6/action-download-artifact
  dependency-version: '11'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump MatteoCampinoti94/changelog-to-release

Bumps [MatteoCampinoti94/changelog-to-release](https://github.com/matteocampinoti94/changelog-to-release) from 1.0.0 to 1.0.6.
- [Release notes](https://github.com/matteocampinoti94/changelog-to-release/releases)
- [Changelog](https://github.com/MatteoCampinoti94/changelog-to-release/blob/main/CHANGELOG.md)
- [Commits](https://github.com/matteocampinoti94/changelog-to-release/compare/v1.0.0...v1.0.6)

---
updated-dependencies:
- dependency-name: MatteoCampinoti94/changelog-to-release
  dependency-version: 1.0.6
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump actions/checkout from 4 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SrBlackVoid committed:
Post-rebase build fixes

SrBlackVoid committed:
Resolving new ScriptAnalzyer tests

For some reason, this time there was a lot more strict formatting rules
applied in ScriptAnalyzer (mostly brace and assignment operator
spacing).

lipkau committed:
fix(tests): fixed failing tests

lipkau committed:
fix(build): updated build debug function

lipkau committed:
Merge pull request #543 from SrBlackVoid/Pester_v5

Updated test suite to Pester v5

lipkau committed:
Merge branch 'master' into dependabot/github_actions/MatteoCampinoti94/changelog-to-release-1.0.6

lipkau committed:
Merge branch 'master' into dependabot/github_actions/dawidd6/action-download-artifact-11

lipkau committed:
Merge branch 'master' into dependabot/github_actions/actions/checkout-6

lipkau committed:
Merge pull request #548 from AtlassianPS/dependabot/github_actions/MatteoCampinoti94/changelog-to-release-1.0.6

⬆️ Bump MatteoCampinoti94/changelog-to-release from 1.0.0 to 1.0.6

lipkau committed:
Merge branch 'master' into dependabot/github_actions/dawidd6/action-download-artifact-11

lipkau committed:
Merge pull request #547 from AtlassianPS/dependabot/github_actions/dawidd6/action-download-artifact-11

⬆️ Bump dawidd6/action-download-artifact from 9 to 11

lipkau committed:
Merge branch 'master' into dependabot/github_actions/actions/checkout-6

lipkau committed:
Merge pull request #546 from AtlassianPS/dependabot/github_actions/actions/checkout-6

⬆️ Bump actions/checkout from 4 to 6

dependabot[bot] committed:
:arrow_up: Bump actions/download-artifact from 5 to 6

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 5 to 6.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump actions/upload-artifact from 4 to 5

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

lipkau committed:
Merge pull request #541 from AtlassianPS/dependabot/github_actions/actions/download-artifact-6

⬆️ Bump actions/download-artifact from 5 to 6

lipkau committed:
Merge branch 'master' into dependabot/github_actions/actions/upload-artifact-5

lipkau committed:
Merge pull request #540 from AtlassianPS/dependabot/github_actions/actions/upload-artifact-5

⬆️ Bump actions/upload-artifact from 4 to 5

dependabot[bot] committed:
:arrow_up: Bump actions/cache from 4 to 5

Bumps [actions/cache](https://github.com/actions/cache) from 4 to 5.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

lipkau committed:
Merge pull request #550 from AtlassianPS/dependabot/github_actions/actions/cache-5

⬆️ Bump actions/cache from 4 to 5

dependabot[bot] committed:
:arrow_up: Bump dawidd6/action-download-artifact from 11 to 12

Bumps [dawidd6/action-download-artifact](https://github.com/dawidd6/action-download-artifact) from 11 to 12.
- [Release notes](https://github.com/dawidd6/action-download-artifact/releases)
- [Commits](https://github.com/dawidd6/action-download-artifact/compare/v11...v12)

---
updated-dependencies:
- dependency-name: dawidd6/action-download-artifact
  dependency-version: '12'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

lipkau committed:
Merge pull request #551 from AtlassianPS/dependabot/github_actions/dawidd6/action-download-artifact-12

⬆️ Bump dawidd6/action-download-artifact from 11 to 12

lipkau committed:
Release v2.15

lipkau committed:
Fixed release labels in CHANGELOG to match deployment logic

lipkau commented:
Thanks @GamerClassN7 — your bug report drove much of the v3 work that has now landed: #569 (search + Cloud detection), #570 (user `accountId`), #573 (ADF helpers), and #583 (paginated createmeta). The follow-up you flagged about worklog and comment bodies needing ADF on Cloud is tracked in #602. Closing as superseded.
